jellyfin.tf

```diff
@@ -202,85 +222,44 @@ resource "aws_instance" "jellyfin_server" {
     ]
   }
 
+  // Creates nginx conf file for jellyfin server based on ALB or Domain name
   provisioner "file" {
-    content = <<EOF
-server {
-  listen 80;
-    server_name ${var.HOSTED_ZONE_ID == "" ? "~^${aws_lb.jellyfin_alb.name}.*\\.elb\\.amazonaws.com$" : trimsuffix(data.aws_route53_zone.jellyfin_domain.0.name,".") };
-  location / {
-    # Proxy main Jellyfin traffic
-    proxy_pass http://localhost:8096/;
-    proxy_set_header Host $host;
-    proxy_set_header X-Real-IP $remote_addr;
-    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
-    proxy_set_header X-Forwarded-Proto $scheme;
-    proxy_set_header X-Forwarded-Protocol $scheme;
-    proxy_set_header X-Forwarded-Host $http_host;
-
-    # Disable buffering when the nginx proxy gets very resource heavy upon streaming
-    proxy_buffering off;
-  }
-  location /socket {
-    # Proxy Jellyfin Websockets traffic
-    proxy_pass http://localhost:8096/socket;
-    proxy_http_version 1.1;
-    proxy_set_header Upgrade $http_upgrade;
-    proxy_set_header Connection "upgrade";
-    proxy_set_header Host $host;
-    proxy_set_header X-Real-IP $remote_addr;
-    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
-    proxy_set_header X-Forwarded-Proto $scheme;
-    proxy_set_header X-Forwarded-Protocol $scheme;
-    proxy_set_header X-Forwarded-Host $http_host;
-  }
-}
-EOF
+    content = templatefile("templates/server.conf.tmpl", { server_name = local.server_name })
```

Comment:
My only concern with this is that the ALB domain names are long. When I tested directly passing in the alb domain name, nginx complained that I needed to increase `server_names_hash_bucket_size`. I suspect that is still the case, and I'd like to keep this working even without a dedicated domain.

Reply:
Fixed the local var for this